### PR TITLE
During Tab Completion, skip file if can't read file attributes

### LINKED ISSUE #7648: Tab completion under /mnt/c stymied by hiberfil.sys
Steps to reproduce
------------------
Running Ubuntu 16.04.4 LTS (via WSL):

```powershell
$error.Clear()
Import-Module /mnt/c/[tab][tab][tab][tab][tab][tab][tab]
#then clear the command line and run
$error
```

Expected behavior
-----------------

```none
No problems, you just get tab completion for stuff under your C drive.
```

Actual behavior
---------------

```none
You don't get any tab completion. For as many times as you hit [tab], you have an error like this in $error:

Exception calling "CompleteInput" with "3" argument(s): "Access to the path '/mnt/c/hiberfil.sys' is denied."
At line:38 char:16
+ ...      return [System.Management.Automation.CommandCompletion]::Complet ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
+ FullyQualifiedErrorId : UnauthorizedAccessException
```

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0-rc.1
PSEdition                      Core
GitCommitId                    6.1.0-rc.1
OS                             Linux 4.4.0-17738-Microsoft #1000-Microsoft Fri Aug 10 17:06:00 PST 2018
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
# PR Summary

During tab completion, while enumerating files, if the file attributes can't be read, just ignore error and skip that file.  Previously, the file access exception wasn't handled and tab completion completely failed without returning any results.

## PR Context

Fix https://github.com/PowerShell/PowerShell/issues/7648

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 1]
```diff
@@ -4275,10 +4275,17 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
...
-                                                    psobjs.Add(wrapper);
+                                                    if ((fileInfo.Attributes & FileAttributes.Hidden) != 0)
+                                                    {
+                                                        PSObject wrapper = PSObject.AsPSObject(entry);
+                                                        psobjs.Add(wrapper);
+                                                    }
+                                                }
+                                                catch
+                                                {
+                                                    // do nothing if can't get file attributes
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: We enumerate files with IgnoreInaccessible = true (line 4251) and this works on Windows but not Unix. It looks like .Net Core issue. Maybe fix there?

Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: The file is enumerated which isn't inaccessible at enumeration time, it fails when reading the file attributes

Comment (iSazonov) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: I don't understand you comment. If the file is not enumerated it is not in the cycle. Really it is enumerated that converse  IgnoreInaccessible = true.

Comment (iSazonov) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: Also the issue for WSL1. Can you check on WSL2?

Comment (iSazonov) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: Looking a history of the issue I think the fix is well for completion.

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs (hunk 1 above)
Comment (SteveL-MSFT) [reply src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs]: I tested on WSL2.  My comment is that the files that cause the exception are enumerated fine.  So it's not a problem with enumeration and IgnoreInaccessible is an EnumerationOption.  The problem is specifically when the file's attributes are being read.  So you have permission to see the file, but not anything about the file.

**Pull request merged**